File: app/src/main/java/com/acrcloud/data/ExternalIds.java

package com.acrcloud.data;

import android.os.Parcel;
import android.os.Parcelable;
import android.os.Parcelable.Creator;
import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class ExternalIds implements Parcelable
{

    @SerializedName("isrc")
    @Expose
    private String isrc;
    @SerializedName("upc")
    @Expose
    private String upc;
    public final static Creator<ExternalIds> CREATOR = new Creator<ExternalIds>() {


        @SuppressWarnings({
            "unchecked"
        })
        public ExternalIds createFromParcel(Parcel in) {
            return new ExternalIds(in);
        }

        public ExternalIds[] newArray(int size) {
            return (new ExternalIds[size]);
        }

    }
    ;

    protected ExternalIds(Parcel in) {
        this.isrc = ((String) in.readValue((String.class.getClassLoader())));
        this.upc = ((String) in.readValue((String.class.getClassLoader())));
    }

    /**
     * No args constructor for use in serialization
     * 
     */
    public ExternalIds() {
    }

    /**
     * 
     * @param isrc
     * @param upc
     */
    public ExternalIds(String isrc, String upc) {
        super();
        this.isrc = isrc;
        this.upc = upc;
    }

    public String getIsrc() {
        return isrc;
    }

    public void setIsrc(String isrc) {
        this.isrc = isrc;
    }

    public String getUpc() {
        return upc;
    }

    public void setUpc(String upc) {
        this.upc = upc;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(isrc);
        dest.writeValue(upc);
    }

    public int describeContents() {
        return  0;
    }

}

File: app/src/main/java/com/acrcloud/data/ExternalIds.kt
package com.acrcloud.data

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class ExternalIds : Parcelable {

    @SerializedName("isrc")
    @Expose
    var isrc: String? = null
    @SerializedName("upc")
    @Expose
    var upc: String? = null

    protected constructor(`in`: Parcel) {
        this.isrc = `in`.readValue(String::class.java.classLoader) as String
        this.upc = `in`.readValue(String::class.java.classLoader) as String
    }

    /**
     * No args constructor for use in serialization
     */
    constructor() {}

    /**
     * @param isrc
     * @param upc
     */
    constructor(isrc: String, upc: String) : super() {
        this.isrc = isrc
        this.upc = upc
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(isrc)
        dest.writeValue(upc)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ExternalIds> {
        override fun createFromParcel(parcel: Parcel): ExternalIds {
            return ExternalIds(parcel)
        }

        override fun newArray(size: Int): Array<ExternalIds?> {
            return arrayOfNulls(size)
        }
    }


}
